pkg/client/cache/reflector_test.go

```diff
@@ -354,11 +355,12 @@ func TestReflectorListAndWatchWithErrors(t *testing.T) {
 func TestReflectorResync(t *testing.T) {
 	iteration := 0
 	stopCh := make(chan struct{})
+	rerr := errors.New("expected resync reached")
 	s := &FakeCustomStore{
 		ResyncFunc: func() error {
 			iteration++
 			if iteration == 2 {
-				close(stopCh)
+				return rerr
```

Comment:
It seems that this error is never returned from [ListAndWatch](https://github.com/kubernetes/kubernetes/blob/8f104a7b0f2298caaf5f8a4ab901855646172886/pkg/client/cache/reflector.go#L336).